parts/linux/cloud-init/artifacts/cse_install.sh

```diff
@@ -200,105 +201,138 @@ downloadSecureTLSBootstrapKubeletExecPlugin() {
     fi
 }
 
-downloadContainerdWasmShims() {
-    declare -a wasmShimPids=()
-    local containerd_wasm_filepath="/usr/local/bin"
-    BOOTSTRAP_PROFILE_CONTAINER_REGISTRY_SERVER="${BOOTSTRAP_PROFILE_CONTAINER_REGISTRY_SERVER:=}"
+wasmFilesExist() {
+    local containerd_wasm_filepath=${1}
+    local shim_version=${2}
+    local version_suffix=${3}
+    shift 3
+    local shims_to_download=("$@")
+
+    local binary_version="$(echo "${shim_version}" | tr . -)"
+    for shim in "${shims_to_download[@]}"; do
+        if [ ! -f "${containerd_wasm_filepath}/containerd-shim-${shim}-${binary_version}-${version_suffix}" ]; then
+            return 1 # file is missing
+        fi
+    done
+    echo "all wasm files exist for ${containerd_wasm_filepath}/containerd-shim-*-${binary_version}-${version_suffix}"
+    return 0 
+}
+
+# Install, download, update wasm must all be run from the same function call
+# in order to ensure WASMSHIMPIDS persists correctly since in bash a new
+# function call from install-dependnecies will create a new shell process.
+installContainerdWasmShims(){
+    local download_location=${1}
+    PACKAGE_DOWNLOAD_URL=${2}
+    shift 2 # shift past the first 2 arguments to capture the list of versions
```

Comment:
Not fully understand why we need the shift here?
Copilot said the {1} (download_location) and {2} (PACKAGE_DOWNLOAD_URL) will be removed. But according to your logic below, it seems not.
Do you mind explaining it a bit more?

Reply:
So everything happening in bash is just strings. Download location and package download url are safely in their variables, and then we “shift” in order to grab the strings in the list and grab them all together. Without the shift it would just grab the first string in the list. 

Chat gpt might be saying that {1} and {2} are no longer accessible which might be the case? But that’s fine since the values are stored in local vars

Reply:
I see I see. Yes they are not accessible but you already stored them.
Copilot gave me another more readible (to me) syntax, just for FYI, no need to change.
```
local download_location=${1}
PACKAGE_DOWNLOAD_URL=${2}
local package_versions=("${@:3}") # Capture all arguments starting from the third one
```

Reply:
I didn't know this syntax existed! I think it looks cleaner as well :)